src/Conn.cc

```diff
@@ -308,9 +308,10 @@ void Connection::RemoveConnectionTimer(double t)
 
 void Connection::SetInactivityTimeout(double timeout)
 	{
-	// We add a new inactivity timer even if there already is one.  When
-	// it fires, we always use the current value to check for inactivity.
-	if ( timeout )
+	// We add a new inactivity timer if there is no current timeout or if the
+	// new timeout is lower than the old one.  When it fires, we always use the
+	// current value to check for inactivity and re-schedule the next timer.
+	if ( timeout && (inactivity_timeout == 0 || timeout < inactivity_timeout) )
 		ADD_TIMER(&Connection::InactivityTimer,
 				last_time + timeout, 0, TIMER_CONN_INACTIVITY);
 
```

Comment:
Should `inactivity_timeout` be set here if the timer doesn't change?

Reply:
yeah, it has to change because Connection::InactivityTimer will pick up the changed value.

If the timeout was 5  minutes and you set it to 10 mins, that means there's already a 5 minute timer running.  When that expires Connection::InactivityTimer will be called and it will re ADD_TIMER  for `last_time + inactivity_timeout` where inactivity_timeout is the new 10 minute value.

If the timeout was 10 minutes and you set it to 5 mins, then you need  to add another timer that fires in 5 minutes.  In this case you do still end up with a redundant timer, but it's a bigger change to loop through `timers` to find and cancel the 10min one that is now obsolete.

Reply:
I will say, this took me an hour to figure out, so I may have gotten it wrong or there may be an even simpler way to fix the redundant timers here.

Reply:
> If the timeout was 10 minutes and you set it to 5 mins, then you need to add another timer that fires in 5 minutes. In this case you do still end up with a redundant timer, but it's a bigger change to loop through `timers` to find and cancel the 10min one that is now obsolete.

Don't think it's too bad to do that loop-and-remove to ensure there's only a single inactivity timer going at any time, so I'll look into adding that during merge (but otherwise this change looked good to me).